R/runs.R

```diff
@@ -366,53 +357,25 @@ delete_tag <- function(key, run_id, client) {
   invisible()
 }
 
-#' Log Parameter
+#' Log Parameters
 #'
-#' Logs a parameter for a run. Examples are params and hyperparams
+#' Logs parameters for a run. Examples are params and hyperparams
 #'   used for ML training, or constant dates and values used in an ETL pipeline.
 #'   A param is a STRING key-value pair. For a run, a single parameter is allowed
 #'   to be logged only once.
 #'
-#' @param key Name of the parameter.
-#' @param value String value of the parameter.
-#' @param run_id A run uuid. Automatically inferred if a run is currently active.
-#' @param client An MLFlow client. Defaults to `NULL` and will be auto-generated.
+#' @inheritParams log_metrics
 #'
 #' @export
-log_param <- function(key, value, run_id, client) {
-
-  stop_for_missing_args(
-    key = maybe_missing(key),
-    value = maybe_missing(value)
-  )
+log_params <- function(..., run_id, client) {
 
-  assert_string(key)
-  assert_string(value)
+  params <- get_key_value_df(...)
```

Comment:
wow, can't believe I hadn't thought about this before, but why not just do this instead of the hacky checking the calling function stuff?
```suggestion
  params <- get_key_value_df(...)
  params$value <- metric_value_to_rest(params$value)
```

Reply:
there was a reason i wasn't doing this, and now i can't remember. this seems so obvious in retrospect

Reply:
@mrkaye97 are we sure this shouldn't be called on metrics, not params? just checking where `metric_value_to_rest` is called in the `master` branch, it's exclusively for metrics and not params

Reply:
yeah I dunno, it seems like it should work okay? `metric_value_to_rest` is vectorized so I figure it'll be fine

Reply:
yeah here are some docs: 

* [log-metric](https://www.mlflow.org/docs/latest/rest-api.html#log-metric) takes a `DOUBLE` for `value`
* [log-param](https://www.mlflow.org/docs/latest/rest-api.html#log-param) takes a `STRING` for `value`

The reason for this is that in the db backend there's a `metrics` table that has a `value` column that's a `DOUBLE`, but in the `parameters` table there same column is `VARCHAR` because you could have a parameter like `engine = "glmnet"`

Reply:
AHHH oops you're right I had it backwards 🤦

Reply:
it looks vectorized, but i get warnings if i call with anything with more than 1 element

```
Warning messages:
1: In if (is.nan(value)) { :
  the condition has length > 1 and only the first element will be used
2: In if (value == Inf) { :
  the condition has length > 1 and only the first element will be used
3: In if (value == -Inf) { :
  the condition has length > 1 and only the first element will be used
```

Reply:
WAIT okay here's the issue: this function is stupidly named.

Reply:
it should be `param_value_to_rest`. I'm not sure what MLFlow will do if you try to give it a metric value that's `Inf`, but we need two separate functions for this. Params need to get cast to strings, so this should be `param_value_to_rest` and metrics could be `NA` or `NaN` or `Inf`, so maybe we should handle that somehow, but definitely separately

Reply:
> it looks vectorized, but i get warnings if i call with anything with more than 1 element
> 
> ```
> Warning messages:
> 1: In if (is.nan(value)) { :
>   the condition has length > 1 and only the first element will be used
> 2: In if (value == Inf) { :
>   the condition has length > 1 and only the first element will be used
> 3: In if (value == -Inf) { :
>   the condition has length > 1 and only the first element will be used
> ```

Oh, lol. I think we need to use `ifelse` 🤦 

Also we should be using `is.infinite()` and not `== Inf`